tools/crd-validation-generator/validation-generator.go

```diff
@@ -89,3 +90,39 @@ func getValidation(filename string) (string, *extv1.CustomResourceValidation) {
 	}
 	return crd.Spec.Names.Singular, crd.Spec.Versions[0].Schema
 }
+
+// sanitizeSchema traverses the given JSON-Schema object and replaces all occurrences of the
+// backtick (`) character in the (sub-)schema Description fields with single quote characters
+func sanitizeSchema(inSchema *extv1.JSONSchemaProps) *extv1.JSONSchemaProps {
+	schema := inSchema.DeepCopy()
+	if schema.Description != "" {
+		schema.Description = strings.ReplaceAll(schema.Description, "`", "'")
+	}
+
+	// Traverse Items
+	if schema.Items != nil {
+		if schema.Items.Schema != nil {
+			schema.Items.Schema = sanitizeSchema(schema.Items.Schema)
+		}
+		if len(schema.Items.JSONSchemas) > 0 {
+			sanitizedProps := make([]extv1.JSONSchemaProps, 0, len(schema.Items.JSONSchemas))
+			for _, schema := range schema.Items.JSONSchemas {
+				sanitizedProps = append(sanitizedProps, *sanitizeSchema(&schema))
+			}
+			schema.Items.JSONSchemas = sanitizedProps
+		}
+	}
+
+	// Traverse Properties
+	if len(schema.Properties) > 0 {
+		propNames := make([]string, 0, len(schema.Properties))
+		for p := range schema.Properties {
+			propNames = append(propNames, p)
+		}
+		for _, name := range propNames {
+			prop := schema.Properties[name]
+			schema.Properties[name] = *sanitizeSchema(&prop)
+		}
+	}
```

Comment:
Why do you need `propNames` slice here? Why not sanitizing the `schema.Properties` map in-place?
```go
// Traverse Properties
for name, prop := range schema.Properties {
    schema.Properties[name] = *sanitizeSchema(&prop)
}
```

Reply:
Good Point!